### PR TITLE
code cleanup:Replace NewRandomVMI calls with libvmi builder pattern

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:
NewRandomVMI* functions are defined in the long tests/utils.go file that is impossible to understand and has repetitious code.
Replace NewRandomVMI calls in functional tests with libvmi builder pattern and Make functional test more resources efficient.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/vm_test.go [hunk 1]
```diff
@@ -1016,12 +1016,12 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			It("Should force stop a VMI", func() {
 
 				By("getting a VM with high TerminationGracePeriod")
-				newVMI := tests.NewRandomVMI()
+				newVMI := libvmi.New(libvmi.WithResourceMemory("32Mi"))
 				gracePeriod := int64(1600)
 				newVMI.Spec.TerminationGracePeriodSeconds = &gracePeriod
```
Comment (dankenigsberg) [inline tests/vm_test.go:1021]: this would be prettier with `libvmi.WithTerminationGracePeriod`

File: tests/vm_test.go [hunk 2]
```diff
@@ -1016,12 +1016,12 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			It("Should force stop a VMI", func() {
```
Comment (dankenigsberg) [inline tests/vm_test.go:1016]: >  Remove newRandomVmi call from vm_test.go and use libvmi instead. 

The commit message would shorter and clearer as `vm_test: replace NewRandomVmi call with libmvi`

File: tests/sysprep_test.go [hunk 3]
```diff
@@ -301,7 +302,7 @@ var _ = Describe("[Serial][Sysprep][sig-compute]Syspreped VirtualMachineInstance
 		libstorage.CreatePVC(OSWindowsSysprep, "35Gi", libstorage.Config.StorageClassWindows, true)
 		answerFileWithKey := insertProductKeyToAnswerFileTemplate(answerFileTemplate)
 		tests.CreateConfigMap("sysprepautounattend", map[string]string{"Autounattend.xml": answerFileWithKey, "Unattend.xml": answerFileWithKey})
-		windowsVMI = tests.NewRandomVMI()
+		windowsVMI = libvmi.New()
 		windowsVMI.Spec = getWindowsSysprepVMISpec()
 		tests.AddExplicitPodNetworkInterface(windowsVMI)
 		windowsVMI.Spec.Domain.Devices.Interfaces[0].Model = "e1000"
```
Comment (dankenigsberg) [inline tests/sysprep_test.go:308]: there's so much more work to be done here...

File: tests/subresource_api_test.go [hunk 4]
```diff
@@ -135,7 +135,7 @@ var _ = Describe("[sig-compute]Subresource Api", func() {
 			})
 
 			It("[test_id:1305][posneg:negative] should return an error when VM is not running", func() {
-				vm := tests.NewRandomVirtualMachine(tests.NewRandomVMI(), false)
+				vm := tests.NewRandomVirtualMachine(libvmi.New(), false)
 				vm, err := virtCli.VirtualMachine(util.NamespaceTestDefault).Create(vm)
 				Expect(err).NotTo(HaveOccurred())
```
Comment (dankenigsberg) [inline tests/subresource_api_test.go:140]: is it the same error as before, or a new one because you did not specify requested memory?

**Review by dankenigsberg — CHANGES_REQUESTED** [review]
the PR text is a bit out of date.

File: tests/subresource_api_test.go (hunk 4 above)
Comment (Barakmor1) [reply tests/subresource_api_test.go:140]: Hey Dan,
It's the same error as before.

File: tests/vm_test.go (hunk 1 above)
Comment (Barakmor1) [reply tests/vm_test.go]: Done,
Thanks!

File: tests/vm_test.go (hunk 2 above)
Comment (Barakmor1) [reply tests/vm_test.go:1016]: Done.

File: tests/sysprep_test.go (hunk 3 above)
Comment (Barakmor1) [reply tests/sysprep_test.go]: Done Please have a look.

File: tests/vm_test.go [hunk 5]
```diff
@@ -1016,12 +1016,12 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
 
 				By("getting a VM with high TerminationGracePeriod")
-				newVMI := tests.NewRandomVMI()
-				gracePeriod := int64(1600)
-				newVMI.Spec.TerminationGracePeriodSeconds = &gracePeriod
-
+				newVMI := libvmi.New(
+					libvmi.WithResourceMemory("32Mi"),
```
Comment (EdDev) [inline tests/vm_test.go:1020]: Based on what you chose 32M? Why not 1M? 2M?

File: tests/vm_test.go [hunk 6]
```diff
@@ -1016,12 +1016,12 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+					libvmi.WithTerminationGracePeriod(int64(1600)),
```
Comment (EdDev) [inline tests/vm_test.go:1021]: I do not think  you need to explicitly convert to `int64`.

File: tests/vm_test.go (hunk 5 above)
Comment (Barakmor1) [reply tests/vm_test.go:1020]: The decision to use 32Mi came from this comment:
https://github.com/kubevirt/kubevirt/pull/8075#discussion_r919005138

File: tests/vm_test.go (hunk 6 above)
Comment (Barakmor1) [reply tests/vm_test.go]: Thanks!
I removed the explicit conversion.

File: tests/vm_test.go [hunk 7]
```diff
@@ -1007,14 +1007,14 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			})
 
 			It("Should force stop a VMI", func() {
-
+				const enoughMemForSafeBiosEmulation = "32Mi"
```
Comment (EdDev) [inline tests/vm_test.go:1010]: nit: How about `minMemForSafeBIOSEmulation` ?

**Review by EdDev — CHANGES_REQUESTED** [review]
[First commit review]

>  vm_test.go: replace NewRandomVMI call with libmvi

These a typo.
How about: `tests, vm: Replace NewRandomVMI with libvmi`

> set memory request to 32Mi to Use less resources
> and be more resources efficiency

`Reduce the memory requirements to the minimum needed for the domain to pass the BIOS stage (32Mi).`

**Pull request closed without merging**